### PR TITLE
/users API for AWS - Task 248

### LINKED ISSUE #248: [Task] Implement /users (api) as a Lambda (AWS) function
**Is your feature request related to a problem? Please describe.**
There's not really a problem at hand, other than wanting to do an interesting assignment.

**Describe the solution you'd like**
I would like an AWS Lambda function to be created that has the same functionality as the Azure Function at [/users](https://nsc-fun-dev-usw2-thursday.azurewebsites.net/api/users).

**Describe alternatives you've considered**
We could just not do this, but that seems like it would serve poorly as an assignment, so we've decided to go ahead with doing it.

### PR DESCRIPTION
**Pull request by lbrown51**
Resolves #248 
- This is code that can be uploaded to aws lambda and contains the functionality of the /users api
  - The lambda function is [HttpTriggerAPIUsers](https://us-west-2.console.aws.amazon.com/lambda/home?region=us-west-2#/functions/HttpTriggerAPIUsers?tab=code)
- To use, you can make calls to [https://mchfd7t6l4.execute-api.us-west-2.amazonaws.com/users](https://mchfd7t6l4.execute-api.us-west-2.amazonaws.com/users)
- It accepts two different methods
  - GET, which takes no parameters
  - and POST, which takes {"firstName": first_name, "lastName": last_name, "email": email}
- It makes use of the API Gateway [nsc-api-dev-usw2-thursday](https://us-west-2.console.aws.amazon.com/apigateway/main/api-detail?api=mchfd7t6l4&region=us-west-2)
  - All of the apis can and should be under this API Gateway.
  - I  created mine as /users - ANY which means it takes any HTTP request to /users and handles it. This is also basically how the Azure api works, so I think it makes sense.
- I don't know that it can be tested locally, but you can certainly test it using Postman
- The environment variables are in the configuration settings for the function!

### CODE REVIEW

**Review by farhadbahrehmandhenry — APPROVED** [review]
the only thing I would suggest and it really doesn't matter that much is changing https://mchfd7t6l4.execute-api.us-west-2.amazonaws.com/users to a clickable link in your documentation.

File: aws-functions/HttpTriggerAPIUsers/lambda_function.py [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import os
```
Comment (nataliagourova) [inline aws-functions/HttpTriggerAPIUsers/lambda_function.py:1]: Thank you for setting up the folder, I will be sure to add my code to aws-functions

File: aws-functions/HttpTriggerAPIUsers/lambda_function.py [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+            logging.info('Attempting to add user...')
+            user_req_body = json.loads(event['body'])
+            new_user_id_http_response = add_user(conn, user_req_body)
+            logging.info('User added successfully!')
+            return new_user_id_http_response
+            
+        else:
+            logging.warn(f'Request with method {method} has been recieved, but that is not allowed for this endpoint')
+            return {
+                'statusCode': 405
```
Comment (nataliagourova) [inline aws-functions/HttpTriggerAPIUsers/lambda_function.py:51]: good use of 405 response

File: aws-functions/HttpTriggerAPIUsers/lambda_function.py [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+            
+    #displays erros encountered when API methods were called
+    except Exception as e:
+        return {
+            'error': f'Error: {str(e)}',
+            'statusCode': 500
+        }
+    
+    finally: 
+        conn.close()
```
Comment (nataliagourova) [inline aws-functions/HttpTriggerAPIUsers/lambda_function.py:62]: good use of connection.close() and commit() methods

**Review by nataliagourova — APPROVED** [review]

**Pull request merged**